### PR TITLE
18 various frontend unittest specs

### PR DESCRIPTION
**Pull request by hk-ret**
- refactor call of getNewConnection
- app component unit tests

### CODE REVIEW

**Comment by hk-ret** [comment]
don't merge this pr. need to resolve some conflicts.

**Comment by hk-ret** [comment]
pr is now ready for review

File: src/app/connected-component/connected.component.ts [hunk 1]
```diff
@@ -46,7 +54,9 @@ export class ConnectedComponent extends BaseComponent implements OnInit, OnDestr
             return;
         }
         console.info(this.componentID, 'is shutting down')
-        this.connectionService.removeConnection(this.componentID);
+        if (this.connected) {
```
Comment (DrDiHai) [inline src/app/connected-component/connected.component.ts:57]: This should also set this.connected to false

**Review by DrDiHai — APPROVED** [review]

File: src/app/connected-component/connected.component.ts (hunk 1 above)
Comment (hk-ret) [reply src/app/connected-component/connected.component.ts:57]: added

**Pull request merged**